evaluation/utils/shared.py

```diff
@@ -50,31 +50,36 @@ def model_dump_json(self, *args, **kwargs):
 class EvalOutput(BaseModel):
     # NOTE: User-specified
     instance_id: str
-    instruction: str
     # output of the evaluation
     # store anything that is needed for the score calculation
     test_result: dict[str, Any]
 
+    instruction: str | None = None
+
     # Interaction info
-    metadata: EvalMetadata
-    history: list[tuple[dict[str, Any], dict[str, Any]]]
-    metrics: dict[str, Any]
+    metadata: EvalMetadata | None = None
+    history: list[tuple[dict[str, Any], dict[str, Any]]] | None = None
+    metrics: dict[str, Any] | None = None
     error: str | None = None
 
     # Optionally save the input test instance
     instance: dict[str, Any] | None = None
 
     def model_dump(self, *args, **kwargs):
         dumped_dict = super().model_dump(*args, **kwargs)
+        # Remove None values
+        dumped_dict = {k: v for k, v in dumped_dict.items() if v is not None}
```

Comment:
I think it is just making the output `dict` slightly more succinct 😅 (eg contain only necessary key)  - but this shouldn't be mandatory - we can remove it if you think that's too much

Reply:
and for the below check: now we allow `Metadata` to be empty (e.g., the case for eval_infer) - so we need this check, otherwise `.model_dump()` will raise issues